### PR TITLE
Toxic Exposure: Add None of These Conditions

### LINKED ISSUE #73360: Toxic Exposure: Add None of These Conditions
## Issue Description
- Move from v1 to v3 checkboxes
- Add the 'I am not claiming any conditions related to toxic exposure' checkbox to the `toxic-exposure-conditions` page
- If Veteran selects one or more conditions and also checks the none checkbox, display error message in mock
- If no checkboxes are selected OR if only none is selected, subsequent TE pages should not display.

## Front end tasks
- Ensure the page is styled according to design spec. 
- ~Ensure the back end data from API is displayed in local and review instance~
- Ensure unit tests are available. 
- Ensure your code changes are covered by E2E tests
- Run axe checks using the Chrome or Firefox browser plugin
- Test for color contrast and color blindness issues
- Zoom layouts to 400% at 1280px width
- Test with 1 or 2 screen readers
- Navigate using the keyboard only

## Acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [ ] Unit tests pass
- [ ] E2E tests cover current code and regression
- [ ] End-to-end tests show 0 violations.
- [ ] ~The data returned matches API response (for given user or scenario)~
- [ ] All axe checks pass
- [ ] All color contrast checks pass
- [ ] All zoom testing passes
- [ ] All keyboard checks pass 
- [ ] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by christinec-fftc**
## Summary
- convert checkbox widget to use web components/checkbox group pattern
- add 'none' of these conditions checkbox
- update util functions and tests for upgraded checkboxes and none checkbox
- uses new `updateUiSchema` feature!
- _(Which team do you work for, does your team own the maintenance of this component?)_
@department-of-veterans-affairs/dbex-trex 

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#73360

## Testing done
- Added unit tests. Performed manual tests for visual verification and accessibility
- _Describe the steps required to verify your changes are working as expected_
1. enable toggle `disability_526_toxic_exposure`
2. start a new, original claim
3. add new conditions
4. verify a checkbox for each new condition is displayed on the `/disability/file-disability-claim-form-21-526ez/toxic-exposure-conditions` page, along with a none checkbox 'I am not claiming any conditions related to toxic exposure'
5. When selecting a condition along with none, verify error displays
6. When selecting at least one condition, verify you are taken to the `gulf-war-hazard-1990` page next
7. When selecting no checkboxes or 'none' checkbox, verify you skip the gulf war page and are taken to the `pow` page

## Screenshots
image

## What areas of the site does it impact?
526ez

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx [hunk 1]
```diff
@@ -42,3 +56,100 @@ export const conditionsDescription = (
...
+      }
+    }
+  }
+}
+ *
+ * @param {object} formData - Full formData for the form
+ * @returns {object} Object with id's for each condition
+ */
+export const makeTEConditionsSchema = formData => {
+  const options = (formData?.newDisabilities || []).map(disability =>
```
Comment (tommasina-va) [inline src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:84]: What happens in the case of an empty array here? Would `disability.condition` need the safe navigation operator too? Curious if this be equivalent? `formData?.newDisabilities?.map(disability => `

 On a related note, I see some tests around null condition list for the toxicExposure spec, but I'm wondering if it'd be worthwhile to add a null condition example to the toxicExposureConditions spec?

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx [hunk 2]
```diff
@@ -42,3 +56,100 @@ export const conditionsDescription = (
...
+
+  options.none = {
+    'ui:title': 'I am not claiming any conditions related to toxic exposure',
+  };
+
+  return options;
+};
+
+export const noneAndConditionError =
+  'You selected a condition, and you also selected “I’m not claiming any conditions related to toxic exposure.” You’ll need to uncheck one of these options to continue.';
```
Comment (tommasina-va) [inline src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:136]: Just a clarity consideration--could "Uncheck one of these options to continue" (without the "You'll need to" part) be more clear?

**Review by tommasina-va — DISMISSED** [review]
Great work, Christine! Love the helpful code comments, screenshots in the PR, and thoughtful tests. 🏆

**Review by micahchiang — DISMISSED** [review]
love the inline example output docs

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx (hunk 1 above)
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:84]: Good questions.

In the case of an empty array, there would be no items to loop through, so line 85 would not execute. That example could be another way of preventing the loop however in some wild chance newDisabilities was a different data structure and so returned truthy, we could be incorrectly trying to use an array function on a different data structure.

The null condition (name) scenario is an odd one. So far I've only seen this happening if user picks 'new' for claim type, adds one or more new conditions and then removes them all. In this scenario there is a validation error on the `/new-disabilities/add` page that will not let the user move to the next page, so not sure how they could get to the `toxic-exposure-conditions` page.  
image

The only way I was able to get to the TE conditions page so far without any new conditions is by using the save in progress dev tool and this is what displayed 
image

I think if we wanted to prevent this scenario, we would need to update the `depends` clause on the form config to prevent access to this page entirely, but again, not clear how this could happen. Feels like a separate ticket?

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx (hunk 2 above)
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:136]: This content had some back and forth with Mike and the CAIA team and this was the finalized text. I imagine it was constructed in this way to use [active voice](https://design.va.gov/content-style-guide/plain-language/use-active-voice). I think if we have feedback we should check if there was a ticket for that effort or we can check with Mike when he is back.

**Comment by christinec-fftc** [comment]
Pushed a commit to fix the e2e tests and it auto dismissed the reviews. Re-requesting pls.

**Review by micahchiang — APPROVED** [review]

**Review by tommasina-va — APPROVED** [review]
LGTM!
